### PR TITLE
Update Documentation

### PR DESCRIPTION
**Pull request by lkiesow**
This patch fixes the documentation, explaining how to build and run
Tobira since the old instructions lead to build failures due to missing
steps and incorrect order.

### CODE REVIEW

File: frontend/README.md
Comment (JulianKniephoff) [inline frontend/README.md:15]: Should we mention the other NPM tasks here? Linting? Type-checking? I guess the question is: Is this a guide to build this once for a deployment, or is it for developers, as well?

File: README.md
Comment (JulianKniephoff) [inline README.md:22]: Why remove the hint towards `rustup`? :thinking:

File: README.md
Comment (JulianKniephoff) [inline README.md:37]: Why remove this section? :thinking:

File: backend/README.md
Comment (JulianKniephoff) [inline backend/README.md:33]: That's not directly related to this PR, but I just noticed it, because it touches it, so here goes: Should we really duplicate the entire config file in the docs? :thinking:

File: frontend/README.md
Comment (JulianKniephoff) [inline frontend/README.md:1]: This is basically just a rehash of what's in the main README already. Do we really want this? :thinking:

File: README.md
Comment (JulianKniephoff) [inline README.md:17]: This might seem like a radical thing to say, but do we really want to document this? I would hope that no one ever has to go through this. I think we should rather have a script that produces a deployable build for those that really want to build from source. But I guess that's a different topic ...

**Review by JulianKniephoff — APPROVED** [review]
From my comments you might already guess that I have some doubts regarding the docs. Don't get me wrong, I don't think they are bad docs, or that this PR in particular makes them worse or anything! I'm just always really unsure about how much really should go in the docs, because while docs are useful, they are also almost tautologically mostly one thing: Duplication. And that will inevitably lead to them becoming bad docs, if you don't put lots of effort into them.

So yeah, take these comments as food for thought from a docs cynic, not as criticism of this particular PR. The PR obviously fixes things, and while it also removes some things, that doesn't **really** bother me, either.

File: frontend/README.md
Comment (lkiesow) [reply frontend/README.md:15]: Above, you already argued for short docs. My intention (for now) is to tell people how to get started.
I'm not opposed to adding more stuff here later _if_ it's generally important and it's not explaining default tooling.

File: README.md
Comment (lkiesow) [reply README.md:22]: Why don't we explain how to get npm?
It really depends on your system how to best get this.
The important information here is that this is a rust tool and you need cargo for building the backend.
That should be enough information for developers to get started.
We should avoid having unnecessary or distracting information in here.

I made an exception for floof since that's a somewhat obscure tool no one knows about

File: README.md
Comment (lkiesow) [reply README.md:37]: Several reasons:

1. Same as above: I't unnecessary additional information people can easily find out and it's not our task to explain container orchestration here. Thus, I keep that to a minimum.
2. You don't need this for getting started at all. Removing this will not hinder people in getting Tobira built and run
3. This is a step-by-step instruction. Shutting things down is not something you _should_ do here

File: README.md
Comment (LukasKalbertodt) [inline README.md:3]: Does this also show the badge for `master`? Or is that the latest run, which could be any failing PR?

**Review by LukasKalbertodt — COMMENTED** [review]
While this PR is certainly an improvement over the current state of docs, I don't quite agree with all changes. I actually wanted to improve the docs for some time but didn't get to it yet :(

My major disagreement is probably about mentioning "potentially useful but not directly connected to Tobira" information, like how to stop the docker container or how to get npm/Rust. I am a big fan of that information. While I do understand the argument of "this information is not central to this project and can be found elsewhere", I think mentioning that useful information in our guides is a lot more convenient for the reader and makes getting started with this project a lot easier. And I don't really see the harm. If it's about "noise", then sure, we can try to tackle that with `<details>`, for example. There are surely ways to include that information and not distract from the Tobira-specific stuff.

Additionally, I would suggest moving the docs as I don't really like the current structure of the three READMEs. How about this?

- `README.md`: does not explain how to build, but (eventually) just describes the project, the license and links the important docs below.
- `docs/hacking-on-tobira.md`: a guide how to get started to develop on tobira. Explains that `floof` is probably what they want but also explains the build process and what steps to rerun when changing what files (e.g. "changing API - reexport schema").
- `docs/build.md`: A guide for *users* who want to build from source for production use. This is a lot shorter than `hacking` as they just want to build it once. 
- `docs/prerequisites.md`: explains what is required to build, run and/or develop on Tobira. This is linked from `hacking` and `build.md`.

Something like that. What do you think?

Of course, we can use most of the text suggested in this PR for that; it's mostly about a different structure. If you want I can also change the structure, based on this PR.

File: backend/README.md
Comment (lkiesow) [reply backend/README.md:33]: It's not duplicated since we don't yet have a configuration file anywhere and it needs to be documented somewhere outside the code.

File: frontend/README.md
Comment (lkiesow) [reply frontend/README.md:1]: The reason for me keeping this file was that I expect this to evolve in which case it should be reasonable complete when it comes to the important information.

File: README.md
Comment (lkiesow) [reply README.md:17]: It is. So, let's not discuss this here.
For now, we don't have this and I actually expect that people may still want to build only one part.

File: README.md
Comment (lkiesow) [reply README.md:3]: Not necessarily `master` but whatever is the default branch. For Tobira, this is currently `master`

**Comment by lkiesow** [comment]
@LukasKalbertodt, while I'm not necessarily opposed to your idea, I think this makes no sense right now and is a massive waste of time. I suspect that things might still change quite a bit and writing several documentations for different target audiences – including for production use – is effort and money thrown into the bin. It's something we may want to do once we are near a first prototype and things have settled a bit. But not now.

Right now, we can exclusively target developers because we do not have another target audience. This also means that we don't need a complex documentation structure yet. We can worry about that once we need it. We do not lose anything by waiting. To the contrary, we even gain something since we can then better judge what documentation we need and how it should be structured.

> My major disagreement is probably about mentioning "potentially useful but not directly connected to Tobira" information, like how to stop the docker container or how to get npm/Rust.

Right now we are only three developers, and we already use different tools ;-)

Getting hugely popular tools like npm or Cargo is really easy. Getting information about them as well. Throwing additional, often unnecessary information at people is just distracting them, especially if the information is irrelevant to the actual task at hand which is building Tobira. That said, I can add a few links to additional information if you want? That would mean that people who need more information don't have to search but can easily get to a good place to get it. What do you think?

File: backend/README.md
Comment (JulianKniephoff) [reply backend/README.md:33]: Ooh, I didn't even realize that. x)

WHY DON'T WE HAVE THAT, THOUGH?! :scream: 

But okay, sure, that's out of scope for this PR.

File: frontend/README.md
Comment (JulianKniephoff) [reply frontend/README.md:1]: I speculated as much. Still wanted to have mentioned it, I guess. ^_^

File: backend/README.md
Comment (lkiesow) [reply backend/README.md:33]: I'm fine with moving that somewhere else. I just wasn't sure where to. Let's shortly talk about that at the next developers meeting? Moving it should be easy enough.

**Comment by LukasKalbertodt** [comment]
Fair point about waiting before restructuring all the docs! 

> That said, I can add a few links to additional information if you want? That would mean that people who need more information don't have to search but can easily get to a good place to get it. What do you think?

I think I would like that. But it's not important enough to block this PR. We can still add this later. 

So no remaining concerns from my side regarding this PR!

**Comment by lkiesow** [comment]
>> That said, I can add a few links to additional information if you want? […]
>
> I think I would like that. But it's not important enough to block this PR. We can still add this later.

Already did that ;-P

**Review by LukasKalbertodt — APPROVED** [review]
> Already did that ;-P

Woopsie, missed that.

I read through all changes again and I like them now!

We said we *don't* require two approvals to merge, but since @JulianKniephoff commented here already, I'll not merge yet and let him decide.

**Comment by JulianKniephoff** [comment]
As said, my comments aren't really concerns that would block this, I just wanted to evoke some thinking.

In it goes.

**Pull request merged**
